src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -1205,6 +1205,34 @@ TEST_F(HMICapabilitiesTest,
   EXPECT_NE(content_to_save, content_after_update);
 }
 
+TEST_F(
+    HMICapabilitiesTest,
+    SaveCachedCapabilitiesToFile_LanguageIsNotTheSameAsPersisted_SaveNewLanguageToCache) {
+  SetUpLanguageAndLightCapabilitiesExpectation();
+  std::string new_language = "RU_RU";
+  ON_CALL(*(MockMessageHelper::message_helper_mock()),
+          CommonLanguageToString(_))
+      .WillByDefault(Return(new_language));
+
+  hmi_capabilities_->Init(last_state_wrapper_);
+  hmi_capabilities_->set_active_tts_language(hmi_apis::Common_Language::RU_RU);
+  const std::vector<std::string> sections_to_update{hmi_response::language};
+  const std::string content_to_save = "{\"TTS\": {\"language\":\"RU_RU\"}}";
+
+  CreateFile(kHmiCapabilitiesCacheFile);
+  const std::vector<uint8_t> binary_data_to_save(content_to_save.begin(),
+                                                 content_to_save.end());
+  file_system::Write(kHmiCapabilitiesCacheFile, binary_data_to_save);
```

Comment:
@ydementieiev Seems `kHmiCapabilitiesCacheFile` initially contains a language we expect.

Reply:
@mvorobio [changed](https://github.com/LuxoftSDL/sdl_core/pull/38/commits/f71de2e348e264d566c1eb9265f70b6455774d27)